### DISCUSSION TITLE
custom theme

### DISCUSSION THREAD

**Discussion by Nimo-19** [maaslalani/slides#69] [Q&A] [answered]
Hi, 

I really like the idea of this and am testing it right now. 
But somehow it seems i can not get a custom theme to work.
I copied the theme.json into the same folder as my slide and added the folowing to the beginning of my presentation.
```
---
theme: ./theme.json
---
```
But no customization of the theme.json seems to have any effect. I also downloaded the ascii theme from charmbracelet/glamour into theme.json but no changes in the slides.
the builtin themes like `theme: light` seems to work. 

Am I missing something?

**Comment by maaslalani** [comment]
> But no customization of the theme.json seems to have any effect. I also downloaded the ascii theme from charmbracelet/glamour into theme.json but no changes in the slides.

Did you restart `slides` after making the modifications?

I can't reproduce, my theme customization work correctly. Do you have more detail about your issue?

![image](https://user-images.githubusercontent.com/42545625/124203053-d7378400-daa9-11eb-8622-9cdff9ed77ca.png)

**Comment by l-lin** [comment]
Hello,

I also reproduce the same issue with the latest version (0.4.0).
It seems the issue comes from the fact the [`DefaultTheme`](https://github.com/maaslalani/slides/blob/v0.4.0/styles/styles.go#L26) is loading the `theme.json` from the project. It does not take into account the input set in the markdown file.

According to the [golang documentation on embed](https://golang.org/pkg/embed/):

>  Package embed provides access to files embedded in the running Go program.
>
> Go source files that import "embed" can use the //go:embed directive to initialize a variable of type string, []byte, or FS with the contents of files read from the package directory or subdirectories at __compile time__.

---

# Steps to reproduce

- create a `theme.json`
```bash
# just changing the h1 style
cat <<EOF > /tmp/theme.json
{
  "document": {
    "block_prefix": "\n",
    "block_suffix": "\n",
    "color": "252",
    "margin": 2
  },
  "block_quote": {
    "indent": 1,
    "indent_token": "│ "
  },
  "paragraph": {},
  "list": {
    "level_indent": 2
  },
  "heading": {
    "block_suffix": "\n",
    "color": "39",
    "bold": true
  },
  "h1": {
    "prefix": "# ",
    "suffix": " ",
    "color": "#559",
    "bold": true
  },
  "h2": {
    "prefix": "▓▓▓ ",
    "color": "#1cc"
  },
  "h3": {
    "prefix": "▒▒▒▒ ",
    "color": "#29c"
  },
  "h4": {
    "color": "#559",
    "prefix": "░░░░░ "
  },
  "h5": {},
  "h6": {},
  "text": {},
  "strikethrough": {
    "crossed_out": true
  },
  "emph": {
    "italic": true
  },
  "strong": {
    "bold": true
  },
  "hr": {
    "color": "240",
    "format": "\n--------\n"
  },
  "item": {
    "block_prefix": "• "
  },
  "enumeration": {
    "block_prefix": ". "
  },
  "task": {
    "ticked": "[✓] ",
    "unticked": "[ ] "
  },
  "link": {
    "color": "30",
    "underline": true
  },
  "link_text": {
    "color": "35",
    "bold": true
  },
  "image": {
    "color": "212",
    "underline": true
  },
  "image_text": {
    "color": "243",
    "format": "Image: {{.text}} →"
  },
  "code": {
    "prefix": " ",
    "suffix": " ",
    "color": "203",
    "background_color": "236"
  },
  "code_block": {
    "theme": "dracula",
    "margin": 2
  },
  "table": {
    "center_separator": "┼",
    "column_separator": "│",
    "row_separator": "─"
  },
  "definition_list": {},
  "definition_term": {},
  "definition_description": {
    "block_prefix": "\n🠶 "
  },
  "html_block": {},
  "html_span": {}
}
EOF
```

- create the `slides.md`

```bash
cat << EOF > /tmp/slides.md
---
theme: /tmp/theme.json
---

# foo

bar
EOF
```

- executes `slides`: `slides /tmp/slides.md`
- notice the theme has not changed:

![image](https://user-images.githubusercontent.com/5452304/124388509-962ab400-dce3-11eb-9bef-c6a57caa9016.png)

**Reply by maaslalani** [reply]
I will try to reproduce this and fix it soon! Thank you for the very detailed report.

**Reply by maaslalani** [reply]
As a temporary workaround (until I fix this which should be soon), you can edit the default theme and recompile `slides` which should show you the custom theme.

**Comment by maaslalani** [comment, marked as answer]
Solved by @itmecho in https://github.com/maaslalani/slides/pull/72
